feat: supabase auth

sidmorizon opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

* **New Features**
  * Added email-based one-time password (OTP) authentication flow with enhanced verification dialogs.
  * Introduced new authentication testing gallery for developers.

* **Refactor**
  * Updated authentication system architecture and session management.
  * Enhanced error handling in authentication workflows.
  * Improved storage and token persistence mechanisms.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

sidmorizon committed:
feat: supabase auth

sidmorizon committed:
fix: lint

sidmorizon committed:
chore: update Supabase project URL and public API key in consts.ts

sidmorizon committed:
refactor: update user ID references from supabaseUserId to onekeyUserId across components and services

sidmorizon committed:
refactor: streamline authentication hooks and improve error handling

- Updated `useOneKeyAuth` to `usePrimeAuthV2`, consolidating authentication logic and enhancing clarity.
- Improved error handling in email OTP dialogs and authentication processes with user-friendly toast notifications.
- Refactored session fetching in `SupabaseAuthProvider` to ensure loading state is managed correctly.
- Removed unused code and comments for better maintainability.
- Adjusted component imports to reflect the new authentication hook structure.

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/5e6b8a5e-559d-4f94-83cc-554ea84597b5?refs=5a804609a66b5f62cb73be4aef5793c63d06525b&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/5e6b8a5e-559d-4f94-83cc-554ea84597b5/license?refs=5a804609a66b5f62cb73be4aef5793c63d06525b&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

sidmorizon committed:
feat: add EmailOTPDialog component for email verification

- Introduced `EmailOTPDialog` to handle email OTP verification, including countdown timer and resend functionality.
- Integrated the dialog into the `useOneKeyAuth` hook for streamlined email OTP sending and confirmation.
- Updated various components to utilize the new authentication structure, replacing references to the deprecated `useLoginOneKeyId` hook.
- Enhanced user experience with improved error handling and toast notifications for verification feedback.

sidmorizon commented on packages/kit-bg/src/dbs/simple/entity/SimpleDbEntityPrime.ts:
```diff
@@ -6,17 +6,21 @@ export interface ISimpleDBPrime {
   authToken: string;
 }
 
+// TODO move supabase Storage here
```
和 supabase Storage 合并

sidmorizon commented on packages/kit/src/components/OneKeyAuth/supabase/SupabaseAuthProvider.tsx:
```diff
@@ -0,0 +1,97 @@
+  // const [profile, setProfile] = useState<any>();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  // TODO move to OneKeyAuthGlobalEffects
```
OneKeyAuthGlobalEffects

sidmorizon committed:
refactor: reorganize Supabase storage implementation and remove deprecated files

- Deleted old Supabase storage files and constants, consolidating them into a new shared storage module.
- Updated `getSupabaseClient` to utilize the new shared `supabaseStorageInstance`.
- Enhanced `SimpleDbEntityPrime` to retrieve authentication tokens from the new Supabase storage structure.
- Introduced new constants and utility functions for Supabase project configuration in the shared storage module.

sidmorizon committed:
refactor: enhance SupabaseStorage with caching mechanism for item retrieval

- Introduced a caching layer in the `getItem` method to improve performance and reduce redundant storage access.
- Added utility functions for caching and timing management.
- Updated `setItem` and `removeItem` methods to clear the cache upon modifications, ensuring data consistency.

sidmorizon committed:
refactor: extend cache duration in SupabaseStorage for improved performance

- Increased the maxAge for cached items from 10 seconds to 30 seconds in the SupabaseStorage class, enhancing data retrieval efficiency.

sidmorizon committed:
feat: add AuthGallery navigation and settings integration

- Introduced navigation to the AuthGallery component within the Bootstrap function, enhancing user flow.
- Added a new section in DevSettings for accessing the AuthGallery modal.
- Updated routing configuration to include the AuthGallery modal, ensuring proper navigation handling.
- Adjusted default selected tab in dev settings for improved user experience.

sidmorizon committed:
refactor: enhance useOneKeyAuth structure and improve storage handling

- Renamed `useOneKeyAuth` to `useOneKeyAuthMethods` for clarity and better organization of authentication methods.
- Consolidated authentication logic within `useOneKeyAuth` to streamline the interface and improve readability.
- Added new storage handling tests in `AuthGallery` to facilitate item setting, getting, and removal, enhancing the developer experience.
- Updated `SupabaseStorage` to improve secure storage checks and item retrieval methods, ensuring better performance and reliability.

sidmorizon committed:
feat: add test.txt and update SupabaseStorage with development environment note

- Added a new file `test.txt` to the repository.
- Updated `SupabaseStorage` to include a comment clarifying that secure storage on desktop in development does not function as expected, where values are always empty despite having keys.

sidmorizon committed:
refactor: update README.md to remove deprecated Privy references and clarify Supabase integration

- Removed outdated sections related to Privy user management and batch creation from the README.md.
- Updated TODO list to reflect current tasks, including changes to SupabaseAuthContext and implementation of isReady in supabase hooks.
- Enhanced clarity on the transition from Privy to Supabase for user authentication.

sidmorizon committed:
fix: improve logout process in useOneKeyAuthMethods

- Enhanced the logout function to handle errors gracefully by catching exceptions during the sign-out process.
- Added calls to clear the Supabase storage instance, ensuring a complete logout experience without disrupting the flow.
- This update aims to improve reliability and user experience during authentication state changes.

sidmorizon committed:
chore: update dependencies in package.json and yarn.lock

- Upgraded @metamask/eth-sig-util from version 5.1.0 to 8.2.0 for improved functionality.
- Added new dependencies for @ethereumjs/common and @ethereumjs/tx with their respective versions.
- Updated @metamask/abi-utils and @metamask/utils to the latest versions, enhancing compatibility and performance.
- Made various adjustments to existing dependencies to ensure consistency and stability across the project.

sidmorizon commented:
@coderabbitai resolve

sidmorizon committed:
fix: unit test

originalix commented on packages/kit/src/components/OneKeyAuth/supabase/README.md:
```diff
@@ -0,0 +1,177 @@
+# Init Project
```
文档就不提交了

sidmorizon replied:
已删除

originalix reviewed (commented):
`test.txt` 也移除了？

sidmorizon committed:
chore: remove outdated README.md for Supabase integration

- Deleted the README.md file for the Supabase component of OneKeyAuth, which contained deprecated information and TODOs related to Privy user management and authentication processes.
- This cleanup aims to streamline documentation and focus on current integration practices.

sidmorizon committed:
chore: delete test.txt file

- Removed the `test.txt` file from the repository as it is no longer needed. This cleanup helps maintain a tidy project structure.
